packages/block-library/src/button/edit.js

```diff
@@ -168,7 +168,9 @@ function ButtonEdit( props ) {
 
 	function setButtonText( newText ) {
 		// Remove anchor tags from button text content.
-		setAttributes( { text: newText.replace( /<\/?a[^>]*>/g, '' ) } );
+		setAttributes( {
+			text: newText.toString().replace( /<\/?a[^>]*>/g, '' ),
```

Comment:
looks like this code should be updated _anyway_ to parse the HTML, but surely after this change we can avoid the print/parse loop.

Reply:
Yep, it's avoidable. I didn't want to indulge in changing existing logic too much, but this one could be an easier one. It would require adding the `removeFormat` function as a method to the class though, so I wasn't sure whether to keep it for another PR.

Reply:
it would be nice in these places to add a `@todo` note, if others don't hate that, to say "come back and replace this with the new pattern"